addons/base_import/models.py

```diff
@@ -126,6 +153,65 @@ def get_fields(self, cr, uid, model, context=None,
         # TODO: cache on model?
         return fields
 
+    def _file_type_handler(self, file_type, record, options):
+        file_extension = FILE_TYPE_DICT.get(file_type)
+        if file_extension:
+            return getattr(self, '_read_' + file_extension)(record, options)
+        else:
+            raise UserError(_("Not a compatible file format. It should be CSV, XLS, XLSX or ODS file"))
+
+    def _read_xls(self, record, options):
+        if not xlrd:
+            raise UserError(_(" XLRD library not found. XLS/XLSX file would not import."))
+        if xlrd.__VERSION__.split('.') < ['0', '9', '3']:
+            raise UserError(_(" Lower Version of XLRD, You need to install latest version of xlrd."))
```

Comment:
Why specifically checking for version 0.9.3, what is the issue with earlier versions? (and the error message is wrong, the latest version is now 0.9.4)